src/scheduler/Scheduler.cpp

```diff
@@ -470,43 +475,34 @@ Scheduler::getRecordedMessagesShared()
     return recordedMessagesShared;
 }
 
-bool Scheduler::hasHostCapacity()
+void Scheduler::incrementInFlightCount(const faabric::Message& msg)
 {
-    return _hasHostCapacity;
+    const std::string funcStr = faabric::util::funcToString(msg, false);
```

Comment:
Just to double check, this function needs no lock because it only runs in the master right?

Reply:
This function should need no lock as it's only ever called from functions that already have a lock, and it's `private`. There may be two instances of the same function executing on the same host, so locking does still matter.